File: modules/browsemap/src/main/java/com/sygic/modules/browsemap/di/BrowseMapComponent.java
package com.sygic.modules.browsemap.di;

import com.sygic.modules.browsemap.BrowseMapFragment;
import com.sygic.modules.browsemap.di.module.ViewModelModule;
import com.sygic.modules.common.di.ModulesComponent;
import com.sygic.modules.common.di.util.ModuleBuilder;
import dagger.Component;

import javax.inject.Scope;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@Scope
@Retention(RetentionPolicy.RUNTIME)
@interface Browse { }

@Browse
@Component(
        modules = {
                ViewModelModule.class
        },
        dependencies = {
                ModulesComponent.class
        }
)
public interface BrowseMapComponent {
    @Component.Builder
    abstract class Builder implements ModuleBuilder<BrowseMapComponent> {}

    void inject(BrowseMapFragment fragment);
}

File: modules/browsemap/src/main/java/com/sygic/modules/browsemap/di/BrowseMapComponent.kt
package com.sygic.modules.browsemap.di

import com.sygic.modules.browsemap.BrowseMapFragment
import com.sygic.modules.common.di.ModuleBuilder
import com.sygic.modules.common.di.ModulesComponent
import dagger.Component
import javax.inject.Scope


@Scope
@Retention(AnnotationRetention.RUNTIME)
annotation class Browse

@Browse
@Component(
    modules = [
        ViewModelModule::class
    ],
    dependencies = [
        ModulesComponent::class
    ]
)
interface BrowseMapComponent {
    @Component.Builder
    abstract class Builder : ModuleBuilder<BrowseMapComponent>

    fun inject(fragment: BrowseMapFragment)
}